ts/defs.bzl

```diff
@@ -379,6 +379,17 @@ def ts_project(
     if tsc != _tsc and tsc_worker == _tsc_worker:
         supports_workers = 0
 
+    # Disable typescript version detection if tsc is not the default:
+    # - it would be wrong anyways
+    # - it is only used to warn if worker support is configured incorrectly.
+    if tsc != _tsc:
+        is_typescript_5_or_greater = None
```

Comment:
I guess it would be here that we can potentially set `supports_worker = False` and maybe remove some code from within the rule? Does that simplify anything?

Reply:
I don't think this would be correct: It *is* possible to supply a custom that has worker support. But if we set `supports_worker = 0` here, workers would be disabled.

They way I think about this is: `is_typescript_5_or_greater` is a best effort configuration fix: If we can "prove" the config is wrong, we correct it and warn. If we cannot "prove" the config is wrong, we do what it says (and it might fail).